pkg/p2p/libp2p/connections_test.go

```diff
@@ -561,6 +563,140 @@ func TestTopologyOverSaturated(t *testing.T) {
 	waitAddrSet(t, &n2disconnectedPeer.Address, &mtx, overlay1)
 }
 
+func TestWithDisconnectStreams(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	s1, overlay1 := newService(t, 1, libp2pServiceOpts{libp2pOpts: libp2p.Options{
+		FullNode: true,
```

Comment:
not sure if `FullNode` is actually needed here, no? since you're overriding the actual handler, this should not make a difference AFAIK

Reply:
fixed